### PR TITLE
Improve speed of cipher operations for large inputs

### PR DESCRIPTION
**Pull request by KostasTsiounis**
In cases were bytes remain buffered and/or padding is enabled and encryption is performed, the entirety of the input provided in the `doFinal()` methods of the `CipherCore` class would be copied in a buffer with enough space to accommodate the extra bytes requires for buffered bytes and/or padding. The buffer would subsequently be zeroed out for security reasons.

To avoid said actions and improve the performance of the process in case of very large inputs, the updated code calls the `update()` method with as much data as possible before the final steps.

Signed-off by: Kostas Tsiounis [kostas.tsiounis@ibm.com](mailto:kostas.tsiounis@ibm.com)

### CODE REVIEW

File: src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java [hunk 1]
```diff
@@ -813,17 +827,8 @@ int doFinal(byte[] input, int inputOffset, int inputLen, byte[] output,
                 int outputOffset)
         throws IllegalBlockSizeException, ShortBufferException,
                BadPaddingException {
-
```
Comment (keithc-ca) [inline src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java:816]: Please don't make unnecessary changes relative to upstream. Please restore this blank line and split line 831 back in two.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java (hunk 1 above)
Comment (KostasTsiounis) [reply src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java]: Restored to original.

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
Pls hold off on merging this until at least tomorrow.

**Comment by keithc-ca** [comment]
Jenkins test sanity.openjdk alinux64 jdk20

**Comment by pshipton** [comment]
This is too late for 20.0.2 so if merged this change will not go into any release.
